dll/ntdll/rtl/version.c

```diff
@@ -93,7 +93,7 @@ SetRosSpecificInfo(IN OUT PRTL_OSVERSIONINFOEXW VersionInformation)
  * @implemented
  */
 BOOLEAN NTAPI
-RtlGetNtProductType(PNT_PRODUCT_TYPE ProductType)
+RtlGetNtProductType(_Out_opt_ PNT_PRODUCT_TYPE ProductType)
```

Comment:
Why `_Out_opt_` ?

Reply:
`opt` stands for optional.

Reply:
Yes I know, but clearly the ProductType cannot be "optional" here...

Reply:
Because `NULL` is not a special value for this function: the latter works fine with valid `0x0` address.

Reply:
No idea whether it actually works fine when being tested on Windows, but definitively it doesn't with the current code in ReactOS.

Reply:
(I am working on a Wine-test, separately: we'll see when it reaches ReactOS.)

`_opt_`: I re-read some different explanations. I am still unclear about how it works on the caller.
Nonetheless, it seems to mean the callee must validate the value, which this function does not.
So I'll remove it...